packages/compiler/src/core/program.ts

```diff
@@ -244,18 +245,24 @@ export async function compile(
     program.reportDiagnostics(await linter.extendRuleSet(options.linterRuleSet));
   }
 
+  initializeSpinner("Checker...");
   program.checker = createChecker(program, resolver);
   program.checker.checkProgram();
+  stopSpinner();
```

Comment:
I think also the whole compiler step can be combine into one, we just seperate out the emitters